homework6/counter.py

```diff
@@ -0,0 +1,46 @@
+"""
+Написать декоратор instances_counter, который применяется к любому классу
+и добавляет ему 2 метода:
+get_created_instances - возвращает количество созданых экземпляров класса
+reset_instances_counter - сбросить счетчик экземпляров,
+возвращает значение до сброса
+Имя декоратора и методов не менять
+
+Ниже пример использования
+"""
+
+
+def instances_counter(input_cls):
+    """Some code"""
+    input_cls.counter = 0
+    new = input_cls.__new__
+
+    def __new__(cls):
+        cls.counter += 1
+        return new(cls)
```

Comment:
You have a good thought here - save a link to an old `__new__` method to be able to call it latter. It could be achieved using `super`.

Reply:
I don't pretty understand how correctly call `super()` inside the function

Reply:
`super().__new__(cls)` will call `__new__` method of the parent class (it is actually a bit complicated because of multiple inheritance), so you don't need to save it. There is a pretty good talk from python core contributor about it: https://youtu.be/EiOglTERPEo

Reply:
Done